packages/salesforcedx-vscode-soql/src/queryDataView/queryDataHtml.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as fs from 'fs';
+import * as path from 'path';
+import { Uri, Webview } from 'vscode';
+import { DATA_VIEW_UI_PATH, HTML_FILE } from '../constants';
+import { HtmlUtils } from '../editor/htmlUtils';
+
+export function getHtml(
+  assets: { [index: string]: Uri },
+  extensionPath: string,
+  webview: Webview
+): string {
+  const {
+    baseStyleUri,
+    tabulatorStyleUri,
+    viewControllerUri,
+    tabulatorUri
+  } = assets;
+
+  const pathToDataViewDist = path.join(extensionPath, DATA_VIEW_UI_PATH);
+  const pathToHtml = path.join(pathToDataViewDist, HTML_FILE);
+  let html = fs.readFileSync(pathToHtml).toString();
+  /*
+  We need to replace the hrefs with webviewUris,
+  this will need to change once we need a standalone data view.
+   */
+  html = HtmlUtils.replaceCspMetaTag(html, webview);
+  html = html.replace('${tabulatorStyleUri}', tabulatorStyleUri.toString());
+  html = html.replace('${baseStyleUri}', baseStyleUri.toString());
+  html = html.replace('${tabulatorUri}', tabulatorUri.toString());
+  html = html.replace('${viewControllerUri}', viewControllerUri.toString());
+
+  return html;
```

Comment:
Do we need to take care of sanitizing the document to ensure it's safe to load?

Reply:
Well there are 
Content Security Protocols in place on the document to only load scripts from `self` 
```
const cspMetaTag = `<meta
      http-equiv="Content-Security-Policy"
      content="default-src 'self';
      img-src ${webview.cspSource} https:;
      script-src ${webview.cspSource};
      style-src 'unsafe-inline' ${webview.cspSource};"
    />`;
```
As well as restrictions on the webview to only load assets from a specified source
```
localResourceRoots: [
          vscode.Uri.file(
            path.join(
              QueryDataViewService.extensionPath,
              DATA_VIEW_RESOURCE_ROOTS_PATH
            )
          )
        ],
```
There is no user input in the webview.
I'm not sure if that covers our concerns or not?

Reply:
Yeah not entirely sure either, might be worth reaching out to Nick. Like in this case even if there is no user input, what if the document at `const pathToHtml = path.join(pathToDataViewDist, HTML_FILE);` was replaced with a malicious one and then loaded here `let html = fs.readFileSync(pathToHtml).toString()`

Reply:
@brpowell I think that should be less of our concerns. If a malicious actor has access to the html file, it also has access to the extension executable JavaScript files, basically we couldn't guarantee anything we are checking here. CSP and no user input should be fine for our use cases here.

Reply:
Thanks for mentioning it Bryan and for the input @xyc, a lot of the consideration for vs code webview stuff is new to me as well.